src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,223 @@
+    const customList: JSX.Element[] = [];
+    const GenerateCustomChild = (): any => {
+        let counter = 0;
+        if (customChildPropsForTesting.length > 0) {
+            for (
+                let i: number = 0;
+                i < customChildPropsForTesting.length;
+                i++
+            ) {
+                counter++;
+                if (customChildPropsForTesting[i]) {
+                    customList.push(
+                        <DivTestItem
+                            key={counter}
+                            style={{
+                                gridArea: `${customChildPropsForTesting[i][0]} / ${customChildPropsForTesting[i][1]} / ${customChildPropsForTesting[i][2]} / ${customChildPropsForTesting[i][3]}`,
+                            }}
+                        >
+                            <CodeStyled>
+                                gridArea({counter}):
+                                {customChildPropsForTesting[i][0]}/
+                                {customChildPropsForTesting[i][1]}/
+                                {customChildPropsForTesting[i][2]}/
+                                {customChildPropsForTesting[i][3]}
+                            </CodeStyled>
+                        </DivTestItem>,
+                    );
+                }
+            }
+        }
+    };
+    // list array store the elements generated based on the grid size
+    // Elements in list are genreated using GenerateChild function
+    // list is generated when drawChildren is true
+    const list: JSX.Element[] = [];
+    const GenerateChild = ({
+        col,
+        row,
+        children,
+    }: {
+        col: number;
+        row: number;
+        children?: any;
+    }): any => {
+        let counter = 0;
+        for (let r: number = 1; r <= row; r++) {
+            for (let c: number = 1; c <= col; c++) {
+                counter++;
+                list.push(
+                    <DivTestItem
+                        key={counter}
+                        style={{
+                            gridArea: `${responsive ? 0 : r} / ${
+                                responsive ? 0 : c
+                            } / ${responsive ? 0 : r + 1} / ${
+                                responsive ? 0 : c + 1
+                            }`,
+                        }}
+                    >
+                        <CodeStyled>
+                            gridArea({counter}): {responsive ? 0 : r}/
+                            {responsive ? 0 : c}/{responsive ? 0 : r + 1}/
+                            {responsive ? 0 : c + 1}
+                        </CodeStyled>
+                        {children}
+                    </DivTestItem>,
+                );
+            }
+        }
+    };
+
+    if (responsiveProps?.enableAuto) {
+        return (
+            <DivStyled
+                {...() => {}}
+                style={{
+                    // Grid Auto area css
```

Comment:
Please move all the code in style to styles.tsx file. It would make code much cleaner 😄 
For reference, you can check NewComp component 🙌🏻